optimism/src/main.rs

```diff
@@ -155,9 +155,12 @@ pub fn main() -> ExitCode {
 
         // TODO: unify witness of MIPS to include the instruction and the error
         for i in 0..MIPS_COLUMNS {
-            if i < SCRATCH_SIZE {
+            if i < MIPS_SELECTORS_LENGTH && i != MIPSColumn::Selector(instr).ix() {
+                // Set to zero all selectors except for the one corresponding to the current instruction
+                mips_trace.witness.get_mut(&instr).unwrap().cols[i].push(Fp::zero());
+            } else if i < SCRATCH_SIZE + MIPS_SELECTORS_LENGTH {
```

Comment:
The condition is incorrect. What happens is i = MIPSColumn::Selector(instr).ix()? `i - MIPS_SELECTOR_LENGTH` will underflow.

Reply:
Yep, exactly, that's the conditional I mentioned. Fixing it in a commit that was not pushed yet. I am modifying the code so that these columns can be instantiated more efficiently.